chia/wallet/wallet.py

```diff
@@ -291,6 +291,7 @@ async def _generate_unsigned_transaction(
         memos: Optional[List[bytes]] = None,
         negative_change_allowed: bool = False,
         min_coin_amount: Optional[uint128] = None,
+        exclude_coins: Optional[Set[Coin]] = None,
```

Comment:
why is this not just `Optional[List[Coin]]`? solves a lot of problems

Reply:
This is a set to be consistent with `coins` from above

Reply:
Why dont we just make everything a list?

Reply:
Good question, i found it a bit weird to constantly switch between set and list (both have their up and downsides).
I dont really want to make such "huge" changes, those are probably better suited for a refactoring pr unrelated to this issue that uses set or list consistently across the codebase for coin sets/lists

Reply:
> Good question, i found it a bit weird to constantly switch between set and list (both have their up and downsides). I dont really want to make such "huge" changes, those are probably better suited for a refactoring pr unrelated to this issue that uses set or list consistently across the codebase for coin sets/lists

I meant just for exclude_coins, i think the normal coins should stay a set for efficiency reasons

Reply:
Having different types feels weird to me

Reply:
Unless you think exclude_coins will end up quite large, which then a set makes sense.

Reply:
i mean `coins` and `exclude_coins` are basically the same thing

Reply:
> Having different types feels weird to me

in my mind longer sets are better then longer lists & sets are a bit faster to loop through.

Reply:
> i mean `coins` and `exclude_coins` are basically the same thing

true, but generally exclude_coins will be smaller than coins.
Please don't switch between sets and lists in this pr though, just use lists and this can be a discussion for later

Reply:
I kept the api consistent, `exclude_coins` should be the same type as `coins`, anything else is unexpected.

Reply:
> I kept the api consistent, `exclude_coins` should be the same type as `coins`, anything else is unexpected.

fair, ill refactor it later if need be.